plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeMetadata.java

```diff
@@ -349,11 +349,10 @@ public Object[][] testApplyProjectionProvider()
                         ImmutableSet.of()
                 },
                 {
-                        // table handle has dereference column projections (which should be filtered out)
                         ImmutableSet.of(DOUBLE_COLUMN_HANDLE, BOOLEAN_COLUMN_HANDLE, DATE_COLUMN_HANDLE, BOGUS_COLUMN_HANDLE, VARCHAR_COLUMN_HANDLE),
```

Comment:
We could have an extra set of tests for `io.trino.plugin.deltalake.TestDeltaLakeMetadata#testApplyProjection` because this is the meat of this PR